src/components/common/LegendReferenceLine/LegendReferenceLine.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { Box, Typography } from '@mui/material';
+import { styled } from '@mui/material/styles';
+
+interface LegendReferenceLineProps {
+  name: string;
+  value: React.ReactNode;
+  color: string;
+}
+
+const Container = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  gap: theme.spacing(1),
+}));
+
+// This component shows the color, name, and value of a reference line in a recharts graph. It
+// should be put in the header of the card containing the graph.
+export const LegendReferenceLine: React.FC<LegendReferenceLineProps> = ({
+  name,
+  value,
+  color,
+}) => (
+  <Container>
+    <Box
+      sx={{
+        display: 'inline-block',
+        height: '5px',
+        width: '20px',
+        borderRadius: '5px',
+        backgroundColor: color,
+      }}
```

Comment:
> Also, are we losing the margin right and bottom or are they no longer needed?

`margin-bottom` isn't needed anymore because I added `justify-content` to the parent to center vertically. And `margin-right` isn't needed anymore because I added `gap` to the parent. Thanks for being so thorough!

Reply:
> We could use a styled component here

I'll used a styled component.